Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -119,13 +122,39 @@ void UPSSaveGameData::UnlockAllLevels()
 // Retrieves the progression reward based on the end game state for the current level.
 float UPSSaveGameData::GetProgressionReward(EEndGameState EndGameState)
 {
-	constexpr float DefaultMultiplier = 1.0f;
-	const float DifficultyMultiplier = UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
+	constexpr float ProgressionReward = 0.f;
 
-	const float* LevelReward = CurrentProgressionSettingsRowData.ProgressionEndGameValues.Find(EndGameState);
-	const float ProgressionReward = LevelReward ? *LevelReward : DefaultMultiplier;
-	return ProgressionReward * DifficultyMultiplier;
+	const UCurveTable* DifficultyCurveTable = UPSDataAsset::Get().GetProgressionDifficultyMultiplierCurveTable();
+	if (!ensureMsgf(DifficultyCurveTable, TEXT("ASSERT: [%i] %s:\n'DifficultyCurveTable' is not valid!"), __LINE__, *FString(__FUNCTION__)))
+	{
+		return ProgressionReward;
+	}
+
+	static const UEnum* Enum = StaticEnum<EEndGameState>();
+	const FString ContextString = Enum->GetNameStringByValue(static_cast<int64>(EndGameState));
```

Comment:
Next way to obtain string is outdated in modern unrea:
```
static const UEnum* Enum = StaticEnum<EEndGameState>();
const FString ContextString = Enum->GetNameStringByValue(static_cast<int64>(EndGameState));
```
Instead, just call `FString EnumStr = UEnum::GetValueAsString(EndGameState);`

Reply:
done: https://github.com/moinm3uw/ProgressionSystem/pull/57/commits/b99a2b66ea4c4f8b0999325828c879c1394342a1

This requires to have EEndGameState:: in the curvetable 
![image](https://github.com/user-attachments/assets/f1673df4-d5ba-4e46-9dcf-578644a8e90c)

Reply:
@moinm3uw, you are right, but you still can obtain it in readable for designers format, with modern methods:
```
const FString ContextString  = UEnum::GetDisplayValueAsText(EndGameState).ToString();
```
Result:
```
Honor Loss
Draw
Win
Loose
```

Reply:
@moinm3uw, you are right, but you still can obtain it in readable for designers format, with modern methods:
```
const FString ContextString  = UEnum::GetDisplayValueAsText(EndGameState).ToString();
```
Result:
```
Honor Loss
Draw
Win
Loose
```

Reply:
done: https://github.com/moinm3uw/ProgressionSystem/pull/57/commits/47ceae05d9b63c53821d6793a4121b49e500267b